contrib/ros/envs/sawyer_env.py

```diff
@@ -0,0 +1,346 @@
+                            'object0'][2])),
+                model_path=osp.join(
+                    rospkg.RosPack().get_path('sawyer_sim_examples'), 'models',
+                    'block/model.urdf'))
+            self._gazebosrv.load_gazebo_sdf_model(
+                model_name='target',
+                model_pose=Pose(
+                    position=Point(gripper_pose['position'][0], gripper_pose[
+                        'position'][1], gripper_pose['position'][2])),
+                model_path=osp.join(
+                    rospkg.RosPack().get_path('sawyer_sim_examples'), 'models',
+                    'target/model.sdf'))
+        else:
+            # TODO(gh/4: Finish 'initial_setup' function for real robot)
+            pass
+
+    # yapf: disable
+    def sample_goal(self):
+        """
+        Sample goals
+        :return: the copy of new sampled goal
+        """
+        if self.has_object:
+            goal = self.initial_gripper_pos[:3] + self.np_random_generator.uniform(
+                                                    -self.target_range,
+                                                    self.target_range,
+                                                    size=3)
+            goal[2] = self.height_offset
+            # If the target can't set to be in the air like pick and place task
+            # it has 50% probability to be in the air
+            if self.target_in_the_air and self.np_random_generator.uniform(
+            ) < 0.5:
+                goal[2] += self.np_random_generator.uniform(0, 0.45)
+        else:
+            goal = self.initial_gripper_pos[:3] + self.np_random_generator.uniform(
+                                                    -0.15, 0.15, size=3)
+        return copy.deepcopy(goal)
+    # yapf: enable
+
+    # -------------------------------------------------------------------------------------------
+    def _get_obs(self):
+        robot_obs = self._robot.get_obs()
+
+        # gazebo data message
+        if self.has_object:
+            object_pos = np.array([])
+            object_ori = np.array([])
+            if self.model_states is not None:
+                model_names = self.model_states.name
+                try:
+                    object_idx = model_names.index('block')
+                    object_pose = self.model_states.pose[object_idx]
+                    object_pos = np.array([
+                        object_pose.position.x, object_pose.position.y,
+                        object_pose.position.z
+                    ]).copy()
+                    object_ori = np.array([
+                        object_pose.orientation.x, object_pose.orientation.y,
+                        object_pose.orientation.z, object_pose.orientation.w
+                    ]).copy()
+                except ValueError as err:
+                    logger.log(
+                        'Error _get_obs(484): The model named "block" does not exist!'
```

Comment:
Yet another reason not to include line numbers and function names in errors: it is already wrong.